docs/migration/01_migration_guide_0.5.ipynb

```diff
@@ -0,0 +1,884 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "interracial-yellow",
+   "metadata": {},
+   "source": [
+    "# Qiskit Machine Learning v0.5 Migration Guide\n",
+    "\n",
+    "This tutorial will guide you through the process of migrating your code from Qiskit Machine Learning v0.4 to v0.5.\n",
+    "\n",
+    "## Introduction\n",
+    "\n",
+    "The main focus of the 0.5 release of Qiskit Machine Learning is the migration of the base computational blocks like quantum kernels and quantum neural networks to the primitives introduced in Qiskit as well as extended support of the primitives in the algorithms.\n",
+    "\n",
+    "Contents:\n",
+    "\n",
+    "- Overview of the primitives\n",
+    "- High level algorithms/primitives\n",
+    "- New quantum kernel\n",
+    "- New quantum neural networks\n",
+    "- Notable deprecation\n",
+    "\n",
+    "\n",
+    "## Overview of the primitives\n",
+    "\n",
+    "The core capability of quantum computers that sets them apart from from classical computers is their ability to generate non-classical probability distributions at their outputs. The native operations that one can do with a probability distribution is to sample from it or to estimate quantities on it. Consequently, these operations of sampling and estimating form the fundamental building blocks of quantum algorithm development. Thus, as it was [announced](https://research.ibm.com/blog/qiskit-runtime-for-useful-quantum-computing), two basic primitives were introduced, Sampler and Estimator, respectively, that implement these two operations:\n",
+    "\n",
+    "- Sampler class calculates probabilities or quasi-probabilities of bitstrings from quantum circuits. The base class is [qiskit.primitives.BaseSampler](https://qiskit.org/documentation/stubs/qiskit.primitives.BaseSampler.html).\n",
+    "- Estimator class estimates expectation values of quantum circuits and observables. The base class is [qiskit.primitives.BaseEstimator](https://qiskit.org/documentation/stubs/qiskit.primitives.BaseEstimator.html).\n",
+    "\n",
+    "Qiskit Terra provides two implementations:\n",
+    "\n",
+    "- The reference implementation that is statevector based. This implementation does require a backend or a simulator, it relies on the classes from the [quantum_info](https://qiskit.org/documentation/apidoc/quantum_info.html) package.\n",
+    "- The backend based primitives. This implementation requires an instance of a backend to be passed to the primitives.\n",
```

Comment:
Really the backend based primitives are to support provider/backends that do not support primitives directly. For Aer simulator the Aer primitives should be used and IBM devices the runtime primitives. Maybe we can mention this otherwise users not so familiar may not know about the others and assume perhaps the above are all.

Reply:
Added more information on Aer and Runtime primitives.